parso/tree.py

```diff
@@ -191,17 +192,17 @@ def __init__(self, value, start_pos, prefix=''):
         :py:func:`str` Typically a mixture of whitespace and comments. Stuff
         that is syntactically irrelevant for the syntax tree.
         '''
-        self.parent = None
+        self.parent = None  # type: Optional[BaseNode]
```

Comment:
We should probably use variable annotations. Python < 3.6 is not supported anymore.

Reply:
Aha, I'd forgotten that was gone now. 1139e53